db-engine/src/main/java/com/splicemachine/db/iapi/sql/conn/LanguageConnectionContext.java

```diff
@@ -1491,4 +1492,16 @@ void setDB2VarcharCompatibilityModeNeedsReset(boolean newValue,
 
     boolean isCompilingFromTableTempTrigger();
 
+    void addUserJarsToSparkContext();
+
+    boolean isSparkJob();
+
+    void setSparkContext(SparkContext sparkContext);
+
+    SparkContext getSparkContext();
```

Comment:
We should try to reduce direct dependencies on Spark in "core" modules such as db-engine.

Reply:
@dgomezferro OK, I converted this to class Object and moved Spark class and API references to the spark_sql module.